src/Microsoft.OData.Core/StreamExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="StreamExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+    using System;
+    using System.IO;
+
+    /// <summary>
+    /// Extension methods to Stream for .NET35.
+    /// </summary>
+    internal static class StreamExtensions
+    {
+#if !PORTABLELIB
+        /// Extension method to copy one stream to another
+        /// <param name="source">Stream to copy from</param>
+        /// <param name="target">Stream to copy to</param>
+        internal static void CopyTo(this Stream source, Stream target)
+        {
+            int BUFFER_SIZE = 81920;
+            byte[] buffer = new byte[BUFFER_SIZE];

```

Comment:
>new byte[BUFFER_SIZE]; [](start = 28, length = 22)

I don't know how frequently does this method call. If it's high, it maybe another performance problem?

Reply:
This is in the .NET 3.5, where streams don't have a CopyTo method, but still probably better to allocate the buffer just once.

---
In reply to: [268362716](https://github.com/OData/odata.net/pull/1204#discussion_r268362716) [](ancestors = 268362716)

Reply:
Made the buffer static and allocated on first use.

---
In reply to: [268671381](https://github.com/OData/odata.net/pull/1204#discussion_r268671381) [](ancestors = 268671381,268362716)